test/powershell/Language/Classes/Scripting.Classes.NoRunspaceAffinity.Tests.ps1

```diff
@@ -0,0 +1,50 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+Describe "Class can be defined without Runspace affinity" -Tags "CI" {
+
+    It "Applying the 'NoRunspaceAffinity' attribute make the class not affiliate with a particular Runspace/SessionState" {
+        [NoRunspaceAffinity()]
+        class NoAffinity {
+            [string] $Name;
+
+            static [int] Echo() {
+                return [runspace]::DefaultRunspace.Id
+            }
+
+            [int] SetAndEcho([string] $value) {
+                $this.Name = $value
+                return [runspace]::DefaultRunspace.Id
+            }
+        }
+
+        $t = [NoAffinity]
+        $o = [NoAffinity]::new()
+
+        ## Running directly should use the current Runspace/SessionState.
+        $t::Echo() | Should -Be $Host.Runspace.Id
+        $o.SetAndEcho('Blue') | Should -Be $Host.Runspace.Id
+        $o.Name | Should -Be 'Blue'
+
+        ## Running in a new Runspace should use that Runspace and its current SessionState.
+        try {
+            $ps = [powershell]::Create()
+            $ps.AddScript('function CallEcho($type) { $type::Echo() }').Invoke() > $null
```

Comment:
Maybe consider adding a test for a custom ctor here, though probably not needed.

Reply:
A test was added for custom constructor.